.github/workflows/push.yml

```diff
@@ -88,3 +99,41 @@ jobs:
             -d opcache.jit_buffer_size=16M
       - name: Verify generated files are up to date
         uses: ./.github/actions/verify-generated-files
+  WINDOWS:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - x64: true
+            zts: false
+          - x64: false
+            zts: true
+    name: "WINDOWS_${{ matrix.x64 && 'X64' || 'X86' }}_${{ matrix.zts && 'ZTS' || 'NTS' }}"
+    runs-on: windows-2019
+    env:
+      APPVEYOR_REPO_BRANCH: "${{ github.ref_name }}"
+      PHP_BUILD_CACHE_BASE_DIR: c:\build-cache
+      PHP_BUILD_OBJ_DIR: c:\obj
+      PHP_BUILD_CACHE_SDK_DIR: c:\build-cache\sdk
+      PHP_BUILD_SDK_BRANCH: php-sdk-2.2.0
+      PHP_BUILD_CRT: vs16
+      PLATFORM: ${{ matrix.x64 && 'x64' || 'x86' }}
+      THREAD_SAFE: "${{ matrix.zts && '1' || '' }}"
+      INTRINSICS: "${{ matrix.zts && 'AVX' || '' }}"
+    steps:
+      - name: git checkout 1/2
+        run: git config --global core.autocrlf false && git config --global core.eol lf
+      - name: git checkout 2/2
+        uses: actions/checkout@v2
+      - name: Setup MySQL
+        run: |
+          choco install mysql -y --params "/port:3307"
+          mysql --port=3307 --user=root --password="" -e "ALTER USER 'root'@'localhost' IDENTIFIED BY 'mysql_pass!'; FLUSH PRIVILEGES;"
```

Comment:
Consider to use https://github.com/ankane/setup-mysql instead of chocolatey. (Note that there is also https://github.com/ankane/setup-sqlserver, but I haven't used that so far.)

Reply:
Yes, I considered, but given the fact the action still download the files the same way (as another Docker container cannot be launched from Windows in GH Actions) and any configuration still requires running CLI client (the image you posted even for password), I am happy with the way I solved it.